lib/segment/src/index/struct_payload_index.rs

```diff
@@ -338,12 +334,14 @@ impl PayloadIndex for StructPayloadIndex {
 
             Box::new(matched_points)
         } else {
-            let points_iterator_ref = self.points_iterator.borrow();
+            let points_iterator_ref = self.id_tracker.borrow();
             let struct_filtered_context = self.struct_filtered_context(query);
 
             // CPU-optimized strategy here: points are made unique before applying other filters.
             // ToDo: Implement iterator which holds the `visited_pool` and borrowed `vector_storage_ref` to prevent `preselected` array creation
-            let mut visited_list = self.visited_pool.get(points_iterator_ref.max_id() as usize);
+            let mut visited_list = self
+                .visited_pool
+                .get(points_iterator_ref.max_id() as usize + 1);
```

Comment:
Is this a bug fix? I can't notice any other logic changes

Reply:
It is not a bug, cause visited_list is able to auto-adjust. But should give some perf. improvement